traitlets/config/application.py

```diff
@@ -464,13 +466,16 @@ def initialize_subcommand(self, subc, argv=None):
         """Initialize a subcommand with argv."""
         subapp,help = self.subcommands.get(subc)
 
-        if isinstance(subapp, six.string_types):
-            subapp = import_item(subapp)
+        if callable(subapp):
+            self.subapp = subapp(self)
```

Comment:
classes are callable, so this check will catch the common case of subapp being an Application and handle it incorrectly.